### ISSUE TITLE
Bug report: Your title here

### ISSUE THREAD

**Issue by nikoskalogridis** [jslint-org/jslint#382]
**Describe the bug**
Wondering what is the violation of using a fart function here

2. Jslint edition:

v2021.12.20

3. Code snippet (if applicable):

```javascript
export default Object.freeze({
    pickOne: ({one}) => one
});
```

**Expected behavior**
Should allow usage of fart function

**Comment by kaizhu256** [comment]
its a long-running bug that jslint crashes with cryptic-warnings when parsing fat-arrows that are anything but toy-case.  the crashing-and-cryptic-warning-part should ideally be fixed, when maintainers find the time to do so.

whether destructuring in fat-arrows should be tolerated however, is up to debate.

**Comment by nikoskalogridis** [comment]
Thank you for your quick response.

I cannot think of any issues with destructuring a parameter on a fart function but if that is to be avoided I guess there should be a mention on the help page.

Probably when jslint gets smarter it might produce a more helpful error message.

**Comment by kaizhu256** [comment]
should be fixed by following 2 pr's:

- pr #385
  - fix bugs related to how jslint detect fat-arrows (sometimes it gives false-positives).

- pr #386
  - jslint already had existing routines to warn against destructuring (which are the ones you find confusing).
  - the old warnings are replaced with something less cryptic:
```js
// old warning
Expected 'function' before '('.
    pickOne: ({one}) => one
Expected '{' and instead saw '=>'.
    pickOne: ({one}) => one

// new warning
Use 'function (...)', not '(...) =>' when arrow functions become too complex.
    pickOne: ({one}) => one
```

![image](https://user-images.githubusercontent.com/280571/154116177-0b204898-cb53-4e74-8f4c-565bc0d49bf0.png)
